### PR TITLE
22114 - add change of address api specs

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#22114

*Description of changes:*

- Change of address API spec examples and endpoint
- Added Success scenario
- Added different responses

Request:

image

Response:

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: docs/business.yaml [hunk 1]
```diff
@@ -287,6 +376,74 @@ paths:
...
+                              addressRegion: BC
+                              addressType: delivery
+                              deliveryInstructions: 
+                              postalCode: V8N 4R7
+                              streetAddress: TEST D 1
+                              streetAddressAdditional:
+                            mailingAddress: 
+                              actions: [
+                                  addressChanged
+                              ]
```
Comment (vysakh-menon-aot) [inline docs/business.yaml:438]: I don't think this is required

Comment (vysakh-menon-aot) [inline docs/business.yaml:438]: I don't think this is required

File: docs/business.yaml [hunk 2]
```diff
@@ -287,6 +376,74 @@ paths:
...
+                              addressType: mailing
+                              deliveryInstructions: 
+                              postalCode: V8N 4R7
+                              streetAddress: TEST M 2
+                              streetAddressAdditional:
+                          registeredOffice: 
+                            deliveryAddress: 
+                              actions: [
+                                  addressChanged
+                              ]
```
Comment (vysakh-menon-aot) [inline docs/business.yaml:426]: I don't think this is required

File: docs/business.yaml [hunk 3]
```diff
@@ -287,6 +376,74 @@ paths:
...
+                              addressRegion: BC
+                              addressType: delivery
+                              deliveryInstructions: 
+                              postalCode: V8N 4R7
+                              streetAddress: TEST D 2
+                              streetAddressAdditional: 
+                            mailingAddress: 
+                              actions: [
+                                  addressChanged
+                              ]
```
Comment (vysakh-menon-aot) [inline docs/business.yaml:413]: I don't think this is required

File: docs/business.yaml [hunk 4]
```diff
@@ -287,6 +376,74 @@ paths:
...
+                        lastModified: '2024-06-09 19:00:08.470883+00'
+                        legalName: KD TEST BAKING BC LIMITED
+                      changeOfAddress:
+                        legalType: BC
+                        offices:
+                          recordsOffice:
+                            deliveryAddress: 
+                              actions: [
+                                  addressChanged
+                              ]
```
Comment (vysakh-menon-aot) [inline docs/business.yaml:401]: I don't think this is required

**Review by vysakh-menon-aot — CHANGES_REQUESTED** [review]
We should not include any parameter which is not in schema

**Review by vysakh-menon-aot — APPROVED** [review]

**Comment by vysakh-menon-aot** [comment]
@ketaki-deodhar I was completely unaware of the use of `actions: [addressChanged]`, which i asked you to remove. Its not in schema but i could see references of this in legal-api (https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/resources/v2/business/business_filings/business_filings.py#L721). Please add that back.

**Comment by ketaki-deodhar** [comment]
> @ketaki-deodhar I was completely unaware of the use of `actions: [addressChanged]`, which i asked you to remove. Its not in schema but i could see references of this in legal-api (https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/resources/v2/business/business_filings/business_filings.py#L721). Please add that back for now.

ahh.... Thanks Vysakh for looking into it in details. Will add it back.

File: docs/business.yaml [hunk 5]
```diff
@@ -239,6 +323,11 @@ paths:
...
...ity': 'RICHMOND', 'addressCountry': 'CA', 'addressRegion': 'BC', 'addressType': 'delivery', 'deliveryInstructions': '', 'postalCode': 'V8N 4R7', 'streetAddress': 'TEST D 1', 'streetAddressAdditional': '', 'mailingAddress': 'actions': ['addressChanged'], 'addressCity': 'RICHMOND', 'addressCountry': 'CA', 'addressRegion': 'BC', 'addressType': 'mailing', 'deliveryInstructions': '', 'postalCode': 'V8N 4R7', 'streetAddress': 'TEST M 1', 'streetAddressAdditional': '' is not valid under any of the given schemas,path:filing],filing:business:foundingDate:2024-07-16T21:20:55.000+00:00,identifier:BC0883110,legalName:0883110 B.C. LTD.,legalType:BC,changeOfAddress:legalType:BC,offices:registeredOffice:deliveryAddress:actions:[addressChanged],addressCity:RICHMOND,addressCountry:CA,addressRegion:BC,addressType:delivery,deliveryInstructions:,postalCode:V8N 4R7,streetAddress:TEST D 1,streetAddressAdditional:,mailingAddress:actions:[addressChanged],addressCity:RICHMOND,addressCountry:CA,addressRegion:BC,addressType:mailing,deliveryInstructions:,postalCode:V8N 4R7,streetAddress:TEST M 1,streetAddressAdditional:,header:certifiedBy:full name,date:2024-07-23,email:no_one@never.get,name:changeOfAddress "
```
Comment (eason-pan-bc) [inline docs/business.yaml:330]: For this one, we have changed it to less 'scary' and shorter version in the latest update.  Consider changing it to
```yaml
rootCause: "errors:<comma delimited list of potential validation errors by json path>"
```

File: docs/business.yaml [hunk 6]
```diff
@@ -287,6 +376,74 @@ paths:
                       courtOrder:
                         fileNumber: '12345'
                         effectOfOrder: planOfArrangement
+              Change-of-address-request:
```
Comment (eason-pan-bc) [inline docs/business.yaml:379]: with the updated naming convention, the example name is all lower case, please refer to the examples in Teams group chat. e.g. `agm-location-change-request`

File: docs/business.yaml [hunk 7]
```diff
@@ -287,6 +376,74 @@ paths:
...
+                summary: Change of address
```
Comment (eason-pan-bc) [inline docs/business.yaml:380]: Title Case, please 😃

File: docs/business.yaml [hunk 8]
```diff
@@ -287,6 +376,74 @@ paths:
...
+                value:
+                    filing:
+                      header:
+                        name: changeOfAddress
+                        certifiedBy: "full name"
+                        email: "no_one@never.get"
+                        date: '2024-07-23'
+                      business:
```
Comment (eason-pan-bc) [inline docs/business.yaml:388]: same as the one above

**Review by eason-pan-bc — COMMENTED** [review]
Also, there is a conflict due to previously merged PRs for the same file, please address it. 😃

**Comment by ketaki-deodhar** [comment]
> Also, there is a conflict due to previously merged PRs for the same file, please address it. 😃

Thanks for the note but I will do it just before merging to make sure I have the latest updates

File: docs/business.yaml [hunk 9]
```diff
@@ -239,6 +322,11 @@ paths:
                   value:
                     errorMessage: API backend third party service error.
                     rootCause: "errors:<comma delimited list of potential validation errors by json path>"
+                change-of-address-missing-records-office-response:
+                  summary: Change of address - Missing records office address response
```
Comment (eason-pan-bc) [inline docs/business.yaml:326]: Forgot to mention, also the naming here, and the `rootCause` can use the shorter version.

Comment (ketaki-deodhar) [reply docs/business.yaml]: this updated now

**Comment by ketaki-deodhar** [comment]
> > Also, there is a conflict due to previously merged PRs for the same file, please address it. 😃
> 
> Thanks for the note but I will do it just before merging to make sure I have the latest updates

File updated with latest updates

File: docs/business.yaml [hunk 10]
```diff
@@ -258,6 +258,89 @@ paths:
                         priority: false
                         status: PENDING
                         submitter: mocked submitter
+                change-of-address-response:
```
Comment (eason-pan-bc) [inline docs/business.yaml:261]: consider adding the status of this response (success/failed) to the example name. e.g. `consent-continuation-out-success-response`

Comment (eason-pan-bc) [reply docs/business.yaml:261]: other than that, looking good 😄

**Review by eason-pan-bc — APPROVED** [review]

**Pull request merged**